pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/patch/DeletePatchTest.java

```diff
@@ -0,0 +1,154 @@
+                        +
+                        ":\"TEST_PARTICIPANT\",\"parent\":\"oncHistoryDetailId\",\"parentId\":2071,\"tableAlias\":\"t\",\"isUnique\":true,"
+                        + "\"realm\":\"osteo2\"}";
+        Patch notTissueDeletePatch = new Gson().fromJson(notDeleteJson, Patch.class);
+        Assert.assertFalse(PatchFactory.isDeletePatch(notTissueDeletePatch));
+    }
+
+    @Test
+    public void testPatchFactoryProduce() {
+        String oncHistoryDeleteJson = "{\"id\":\"123\",\"user\":\"user1\",\"nameValue\":{\"name\":\"oD.deleted\",\"value\":true},"
+                + "\"ddpParticipantId\":\"TEST_PARTICIPANT\",\"parent\":\"participantId\",\"parentId\":16548,\"tableAlias\":\"oD\""
+                + ",\"isUnique\":true,\"realm\":\"SOME_REALM\"}";
+        Patch oncHistoryDeletePatch = new Gson().fromJson(oncHistoryDeleteJson, Patch.class);
+        BasePatch patcher = PatchFactory.makePatch(oncHistoryDeletePatch, null);
+        Assert.assertTrue(patcher instanceof DeleteOncHistoryPatch);
+
+        String tissueDeleteJson =
+                "{\"id\":\"123\",\"user\":\"user1\",\"nameValue\":{\"name\":\"t.deleted\",\"value\":\"1\"},\"ddpParticipantId\":\"TEST_PARTICIPANT\",\"parent\":\"oncHistoryDetailId\",\"parentId\":2071,\"tableAlias\":\"t\",\"isUnique\":true,\"realm\":\"osteo2\"}";
+        Patch tissueDeletePatch = new Gson().fromJson(tissueDeleteJson, Patch.class);
+        patcher = PatchFactory.makePatch(tissueDeletePatch, null);
+        Assert.assertTrue(patcher instanceof DeleteTissuePatch);
+    }
+
+    @Test
+    public void testDeleteOncHistoryAndUniquePatch() {
+        String institutionId = testParticipantUtil.createInstitutionForParticipant(participantId);
+        String medicalRecordId = testParticipantUtil.createMRForParticipant(institutionId);
+        String oncHistoryId = testParticipantUtil.createOncHistoryForParticipant(medicalRecordId);
+        createdOncHistoryIds.add(oncHistoryId);
+        String tissueId = testParticipantUtil.createTissueForParticipant(oncHistoryId);
+        createdTissueIds.add(tissueId);
+        String smIdPk = testParticipantUtil.createSmIdForParticipant(tissueId, "USS_VALUE");
+        createdSmIdPks.add(smIdPk);
+        List<String> participantIds = new ArrayList<>();
+        participantIds.add(participantGuid);
+        OncHistoryDetail oncHistoryDetail = testParticipantUtil.getOncHistoryDetailById(oncHistoryId);
+        Assert.assertFalse(oncHistoryDetail.isDeleted());
+        Tissue tissue = oncHistoryDetail.getTissues().stream().findFirst().get();
+        Assert.assertEquals(String.valueOf(tissue.getTissueId()), tissueId);
+        SmId tissueSmId = tissue.getUssSMID().stream().filter(smId -> String.valueOf(smId.getSmIdPk()).equals(smIdPk)).findAny().get();
+        Assert.assertEquals(String.valueOf(tissueSmId.getTissueId()), tissueId);
+
+        String tissueSkIdUniquePatch = "{\"id\":" + tissueId + ",\"user\":\"" + testParticipantUtil.userEmail +
```

Comment:
You might want to explain what you are doing as you go along, at least in the testing parts. This looks like at this point you are testing patching as upserts rather than deletes?

Reply:
Might be worth breaking some of these longer tests into separate test methods so it's easier to see what the blocks of tests are doing.